code/modules/recycling/conveyor2.dm

```diff
@@ -184,15 +211,24 @@
 	speed_process = 1
 
 
-/obj/machinery/conveyor_switch/New()
-	..()
+/obj/machinery/conveyor_switch/New(newloc, newid)
+	..(newloc)
+	if(!id)
+		id = newid
 	update()
 
 	spawn(5)		// allow map load
 		conveyors = list()
-		for(var/obj/machinery/conveyor/C in world)
+		for(var/obj/machinery/conveyor/C in machines) //This works if it's: `C in world` , but that's bad practice. Best to try to fix it.

```

Comment:
i don't see /obj/machinery/New anywhere (in machinery.dm right?). Do I have to add it in?